lunavl/sdk/base.py

```diff
@@ -0,0 +1,143 @@
+from abc import ABC, abstractmethod
+from typing import Union, Optional, Tuple, Dict, Any
+
+from FaceEngine import DetectionFloat  # pylint: disable=E0611,E0401
+
+from .image_utils.geometry import LANDMARKS, Point, Rect
+
+
+class BaseEstimation(ABC):
+    """
+    Base class for estimation structures.
+
+    Attributes:
+        _coreEstimation: core estimation
+    """
+
+    __slots__ = ("_coreEstimation",)
+
+    def __init__(self, coreEstimation: Any):
+        self._coreEstimation = coreEstimation
+
+    @property
+    def coreEstimation(self):
+        """
+        Get core estimation from init
+        Returns:
+            _coreEstimation
+        """
+        return self._coreEstimation
+
+    @abstractmethod
+    def asDict(self) -> Union[dict, list]:
+        """
+        Convert to a dict.
+
+        Returns:
+            dict from luna api
+        """
+        pass
+
+    def __repr__(self) -> str:
+        """
+        Representation.
+
+        Returns:
+            str(self.asDict())
+        """
+        return str(self.asDict())
+
+
+class Landmarks(BaseEstimation):
+    """
+    Base class for landmarks
+
+    Attributes:
+        _points (Optional[Tuple[Point[float]]]): lazy load attributes, converted to point list core landmarks
+    """
+
+    __slots__ = ["_points", "_coreEstimation"]
+
+    def __init__(self, coreLandmarks: LANDMARKS):
+        """
+        Init
+
+        Args:
+            coreLandmarks (LANDMARKS): core landmarks
+        """
+        super().__init__(coreLandmarks)
+        self._points: Optional[Tuple[Point[float], ...]] = None
+
+    @property
+    def points(self) -> Tuple[Point[float], ...]:
+        """
+        Lazy load of points.
+
+        Returns:
+            list of points
+        """
+        if self._points is None:
+            self._points = tuple(
+                (Point.fromVector2(self._coreEstimation[index]) for index in range(len(self._coreEstimation)))
+            )
+        return self._points
+
+    def asDict(self) -> Tuple[Tuple[int, int], ...]:  # type: ignore
+        """
+        Convert to dict
+
+        Returns:
+            list to list points
+        """
+        pointCount = len(self._coreEstimation)
+        points = self._coreEstimation
+        return tuple(((int(points[index].x), int(points[index].x)) for index in range(pointCount)))
+
+
+class BoundingBox(BaseEstimation):
+    """
+    Detection bounding box, it is characterized of rect and score:
+
+        - rect (Rect[float]): face bounding box
+        - score (float): face score (0,1), detection score is the measure of classification confidence
+                         and not the source image quality. It may be used topick the most "*confident*" face of many.
+    """
+
+    #  pylint: disable=W0235
```

Comment:
W0235 does not exist
http://pylint-messages.wikidot.com/all-codes

Reply:
http://pylint.pycqa.org/en/latest/technical_reference/features.html?highlight=W0235#pylint-checkers-options-and-switches